lightning/src/ln/interactivetxs.rs

```diff
@@ -1506,12 +1499,9 @@ impl InteractiveTxConstructor {
 		for output in &outputs_to_contribute {
 			let new_output = match output {
 				OutputOwned::Single(_tx_out) => None,
-				OutputOwned::SharedControlFullyOwned(tx_out) => {
-					Some((tx_out.script_pubkey.clone(), tx_out.value.to_sat()))
-				},
 				OutputOwned::Shared(output) => {
 					// Sanity check
-					if output.local_owned >= output.tx_out.value.to_sat() {
+					if output.local_owned > output.tx_out.value.to_sat() {
```

Comment:
Let's make sure this is in a separate commit before merging.

Reply:
This is integral part of the removal of `SharedControlFullyOwned`; as before the 0-or-all cases were represented by `SharedControlFullyOwned`, the `Shared` had the check that it cannot be equal. So this change should go with the  `SharedControlFullyOwned`-removal commit.